google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -88,30 +89,43 @@ private AppEngineUtil() {
   }
 
   /**
-   * Creates a list of module deployment sources available for deployment to App Engine. Currently,
-   * all module based sources target the App Engine flexible environment:
+   * Creates a list of module deployment sources available for deployment to App Engine:
    *
-   * <p>Maven based deployment sources are included if there are no App Engine standard artifacts
-   * associated with the same module.
+   * <p>Maven based deployment sources are included for both flexible and standard projects if
+   * applicable.
    *
-   * <p>User browsable jar/war deployment sources are always available.
+   * <p>User browsable jar/war deployment sources are included only if there are no App Engine
+   * standard modules.
    *
    * @return a list of {@link ModuleDeploymentSource}'s
    */
   public static List<ModuleDeploymentSource> createModuleDeploymentSources(
       @NotNull Project project) {
+    AppEngineProjectService projectService = AppEngineProjectService.getInstance();
+    AppEngineAssetProvider assetProvider = AppEngineAssetProvider.getInstance();
+
     List<ModuleDeploymentSource> moduleDeploymentSources = Lists.newArrayList();
-    AppEngineProjectService aeProjectHelper = AppEngineProjectService.getInstance();
+
+    boolean hasStandardModules = false;
 
     for (Module module : ModuleManager.getInstance(project).getModules()) {
+      AppEngineEnvironment environment =
+          projectService.getAppEngineArtifactEnvironment(
+              assetProvider.loadAppEngineStandardWebXml(project, module));
+
       if (ModuleType.is(module, JavaModuleType.getModuleType())
-          && !aeProjectHelper.containsAppEngineStandardArtifacts(project, module)
-          && aeProjectHelper.isJarOrWarMavenBuild(project, module)) {
-        moduleDeploymentSources.add(createMavenBuildDeploymentSource(project, module));
+          && projectService.isJarOrWarMavenBuild(project, module)) {
+        moduleDeploymentSources.add(createMavenBuildDeploymentSource(project, module, environment));
+      }
+
+      if (environment == AppEngineEnvironment.APP_ENGINE_STANDARD) {
+        hasStandardModules = true;
       }
     }
 
-    moduleDeploymentSources.add(createUserSpecifiedPathDeploymentSource(project));
+    if (!hasStandardModules) {
```

Comment:
since its one unified deployment view for the entire project, my thinking was that given these changes (of not offering to deploy a standard project to flex) that we shouldn't mix flex sources in with standard sources. that condition only affects the user specified path source too.